refinery/tool_manager/utils.py

```diff
@@ -11,3 +17,132 @@ class AdminFieldPopulator(admin.ModelAdmin):
     def __init__(self, model, admin_site):
         super(AdminFieldPopulator, self).__init__(model, admin_site)
         self.list_display = [field.name for field in model._meta.fields]
+
+
+def create_tool_definition_from_workflow(workflow_data):
+    """
+    :param workflow_data: dict of annotated data coming from a Galaxy workflow
+    """
+
+    file_relationship = create_nesting(workflow_data["annotation"])
+
+    tool_definition = ToolDefinitionFactory(
+        name=workflow_data["name"],
+        description=workflow_data["annotation"]["description"],
+        tool_type="WORKFLOW",
+        file_relationship=file_relationship
+    )
+
+    for output_file in workflow_data["annotation"]["output_files"]:
+        tool_definition.output_files.add(
+            OutputFileFactory(
+                name=output_file["name"],
+                description=output_file["description"],
+                filetype=FileType.objects.get(
+                    name=output_file["filetype"]["name"]
+                )
+            )
+        )
+
+    # TODO: figure out Parameter/GalaxyParameter stuff
+
+
+def create_nesting(d, fr_store=None):
```

Comment:
@hackdna: Specifically would like some input on the recursive `create_nesting` calls. I tried to be as verbose as I could with comments.